intro-to-debugging/identifying-errors.md

```diff
@@ -0,0 +1,198 @@
+# Identifying Errors with the Stack Trace
+
+## Introduction
+
+When we call `print("Hello World")` in our Python code, the text `Hello World` gets printed somewhere: the terminal.
+
+When we call `print(variable_that_doesnt_exist)` in our Python code, we also get output in the terminal. We get error messages.
+
+Learning about what these error messages are and how to read them will help us debug!
+
+## Learning Goals
+
+- Define Stack Trace (Traceback in Python)
+- Explain how to read error messages in a stack trace
+- Use the stack trace to identify the line of code that causes an error
+
+## Vocabulary and Synonyms
+
+| Vocab | Definition | Synonyms | How to Use in a Sentence
+| --- | --- | --- | ---
+Stack Trace | A report of the active stack frames at a certain point in time during the execution of a program. Reports error messages and details about that error | Traceback | "When debugging that error, we should check the stack trace to see what line of code caused it," "The stack trace is very long, but we can see what test called this method here."
+
+## The Stack Trace Delivers Error Messages
+
+The stack trace is a report of all of the code running at a specific point. The stack trace will report error messages and details about that error, if the program runs into a syntax or runtime error.
+
+For example, let's consider the following situation:
+1. A Python script named `main.py`
+1. A terminal that runs that file
+
+```python
+# main.py
+
+apples_quantity = 888      # this is line 3
+basket_capacity = 0        # this is line 4
+apple_basket_quantity = number_of_apples / basket_capacity    # this is line 5
+```
+
+```bash
+$ python3 main.py
+```
+
+In that same terminal window, we should see the following stack trace:
+
+```bash
+Traceback (most recent call last):
+  File "main.py", line 5, in <module>
+    apple_basket_quantity = number_of_apples / basket_capacity # this is line 5
+ZeroDivisionError: division by zero
+```
+
+Callout: Let's consider briefly **where** we see error messages, and in general all output. Right now, when we run Python code from the terminal, we see the output in the terminal. That's because our computers and terminals have configured a [standard stream for output](https://en.wikipedia.org/wiki/Standard_streams), or an "output channel." The output channel, or standard stream for output, for the terminal is the terminal itself by default.
```

Comment:
```suggestion
```suggestion
### !callout-info
## Terminal
Let's consider briefly **where** we see error messages, and in general all output. Right now, when we run Python code from the terminal, we see the output in the terminal. That's because our computers and terminals have configured a [standard stream for output](https://en.wikipedia.org/wiki/Standard_streams), or an "output channel." The output channel, or standard stream for output, for the terminal is the terminal itself by default.

We introduce this idea briefly because one day, the standard stream may change. :)
### !end-callout
```
```